src/test/regress/sql/single_node.sql

```diff
@@ -41,6 +41,30 @@ SELECT * FROM test ORDER BY x;
 UPDATE test SET y = y + 1 RETURNING *;
 WITH cte_1 AS (UPDATE test SET y = y - 1 RETURNING *) SELECT * FROM cte_1 ORDER BY 1,2;
 
+-- Test upsert with constraint
+CREATE TABLE upsert_test
+(
+	part_key int UNIQUE,
+	other_col int,
+	third_col int
+);
+-- distribute the table and create shards
+SELECT create_distributed_table('upsert_test', 'part_key');
+
```

Comment:
Can possibly set `citus.log_remote_commands to true` to see that these are indeed executed locally. Otherwise the tests probably don't have much of a value